### PR TITLE
Emit declarations for eslint-plugin

### PR DESCRIPTION
**Pull request by mohsen1**
The `createRule` helper is extremely useful for writing rules in TypeScript. I would like to use that for my own ESLint rules. Exporting the types will help a lot.

### CODE REVIEW

File: packages/eslint-plugin/tsconfig.build.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     // specifically disable declarations for the plugin
```
Comment (j-f1) [inline packages/eslint-plugin/tsconfig.build.json]: Can you remove this comment as it’s no longer accurate?

Comment (j-f1) [reply packages/eslint-plugin/tsconfig.build.json]: Actually, removing the two `declaration` options will cause a fallback to the inherited config, which enables them.

**Comment by j-f1** [comment]
Looks like `RuleModule` needs to be exported.

**Comment by mohsen1** [comment]
It's being exported by the module 

https://github.com/typescript-eslint/typescript-eslint/blob/00eae48cd64028238720e239f9ee04807c4311c4/packages/eslint-plugin/typings/ts-eslint.d.ts#L687

I was getting the same error when I copy-pasted this code into my own repo. I didn't fully resolve that issue. Not sure what's going on...

**Comment by mohsen1** [comment]
Closing until we decide how we want to approach this issue: #330

**Pull request closed without merging**